tests/test_model.py

```diff
@@ -1227,6 +1210,33 @@ def test_set_presentation_edition(self):
         license_pool = edition_composite.is_presentation_for
         eq_(license_pool, pool)
 
+class TestLicensePoolDeliveryMechanism(DatabaseTest):
+
+    def test_set_rights_status(self):
+        edition, pool = self._edition(with_license_pool=True)
+        pool.open_access = False
+        lpdm = pool.delivery_mechanisms[0]
+        uri = "http://foo"
+        status = lpdm.set_rights_status(uri)
+        eq_(status, lpdm.rights_status)
+        eq_(uri, status.uri)
+        eq_(False, pool.open_access)
+
+        status2 = lpdm.set_rights_status(uri)
+        eq_(status, status2)
+
+        uri2 = "http://baz"
+        status3 = lpdm.set_rights_status(uri2)
+        assert status != status3
+        eq_(uri2, status3.uri)
+
+        open_access_uri = RightsStatus.GENERIC_OPEN_ACCESS
+        open_access_status = lpdm.set_rights_status(open_access_uri)
+        eq_(open_access_uri, open_access_status.uri)
+        eq_(True, pool.open_access)
+
+        non_open_access_status = lpdm.set_rights_status(uri)
+        eq_(True, pool.open_access)
```

Comment:
It looks like you're verifying the case I don't like--that changing a delivery mechanism's rights from open access to non-open access leaves the pool's open-access status alone. Is this right?